CRM/Contact/BAO/Query.php

```diff
@@ -4734,17 +4734,26 @@ public static function appendAnyValueToSelect($selectClauses, $groupBy, $aggrega
    *  on full_group_by mode, then append the those missing columns to GROUP BY clause
    * keyword to select fields not present in groupBy
    *
-   * @param string $groupBy - GROUP BY clause where missing ORDER BY columns will be appended
+   * @param string|array $groupBy - GROUP BY clause or array of columns where missing ORDER BY columns will be appended if not present
    * @param array $orderBys - ORDER BY sub-clauses
    *
    */
   public static function getGroupByFromOrderBy(&$groupBy, $orderBys) {
     if (!CRM_Utils_SQL::disableFullGroupByMode()) {
       foreach ($orderBys as $orderBy) {
-        $orderBy = str_replace(array(' DESC', ' ASC', '`'), '', $orderBy); // remove sort syntax from ORDER BY clauses if present
+        $orderBy = str_ireplace(array(' DESC', ' ASC', '`'), '', $orderBy); // remove sort syntax from ORDER BY clauses if present
         // if ORDER BY column is not present in GROUP BY then append it to end
-        if (preg_match('/(MAX|MIN)\(/i', trim($orderBy)) !== 1 && !strstr($groupBy, $orderBy)) {
-          $groupBy .= ", {$orderBy}";
+        if (preg_match('/(MAX|MIN)\(/i', trim($orderBy)) !== 1) {
```

Comment:
so change is coping with possibility of non-string $groupBy

Reply:
So the purpose of this if is to find any columns that are aggregated in the  select and not add them to teh group by right?